_posts/2024-02-14-Are-You-Ready-to-Low-Code-Green?-Introduction to-Sustainable-Canvas-App-and-Power-Automate-Development

```diff
@@ -0,0 +1,134 @@
+---
+layout: post
+title: Are You Ready to Low-Code Green? Introduction to Sustainable Canvas App and Power Automate Development
+author: milla.vaananen
+excerpt: >
+   Some insight and ideas how to implement green coding practicisies in Canvas App and Power Automate Development.
+tags:
+ - Green Coding
+ - Canvas App
+ - Power Platform
+ - Power Automate
+---
+
+![Illustartion of how green software divedes to energt effiency, hardware effiency and carbon awarness](/img/2024-Power-Platform/Green%20Sofware%20Principles.png)
+> Picture: [Green Software Foundation](https://learn.greensoftware.foundation/introduction)
+
+The ICT sector (including data centers, communication networks, and user devices) is estimated to account for [around 4-6% of global electricity usage.](https://post.parliament.uk/research-briefings/post-pn-0677/) When discussing CO2 or greenhouse gas emissions, the ICT sector's emissions primarily stem from electricity consumption. As an app developer, I don’t have the power to dictate the type of electricity end-users or data centers use; therefore, I should focus on reducing the energy consumption of the app.
+
+When working with the Power Platform, one might easily think that developers can't do much. The platform takes care of most things like scaling, data savings, and cloud servers. Here, green coding comes into play.
+
+Green coding is a method of coding that aims to minimize the energy required to process lines of code. While there are no studies available on the efficiency of Power FX (if I were to speculate, there might not be much to brag about it, given that high-level coding languages, like JavaScript, are not typically the most efficient), and we lack the ability to select the language in Canvas apps, our attention should be directed towards enhancing the energy efficiency of our apps through alternative means.
```

Comment:
what does it mean to "process lines of code"? does it mean how much overhead the language brings? a single line could be run 1 time or a million times.

Reply:
what is a "Canvas app", perhaps link to some explanation

Reply:
Thanks! 
Added some clarification ( Process lines of codes --> process single line of code), and some links to power platform and canvas app, to provide more context

Reply:
still, what does it mean? `x = x + 1` is obviously much cheaper than `myVeryBigList.sort()`, but both are single lines of code.

Does it mean some calculated average of the efficiency of the language? If there's some link to a good definition, that could be added.

Reply:
Mainly how it is calculated... you state there are now studies on efficiency of "Power FX", but are there studies for other languages? How do they define the efficiency terms. Perhaps a link to some review or further text regarding the topic.

I could find this, https://dl.acm.org/doi/10.1145/3136014.3136031 (paywalled) and another referencing it https://www.efinancialcareers.com/news/2023/06/which-programming-language-uses-the-most-energy

If you have some good link for the efficiency, it would be good for the uninitiated reader.

Reply:
Thanks, Now I Got It!  I think the study from Pereira et al (2017, the link that was in the paywall) is still the best paper there is on this subject. 
There are also some updates done to it.  I'll add link to it in the blog post, where you can read the paper and measuring methods more =) 
https://sites.google.com/view/energy-efficiency-languages/home